sway/input/seat.c

```diff
@@ -368,11 +368,20 @@ static void seat_update_capabilities(struct sway_seat *seat) {
 			caps |= WL_SEAT_CAPABILITY_TOUCH;
 			break;
 		case WLR_INPUT_DEVICE_TABLET_TOOL:
+			caps |= WL_SEAT_CAPABILITY_POINTER;
+			break;
 		case WLR_INPUT_DEVICE_TABLET_PAD:
 			break;
 		}
 	}
 	wlr_seat_set_capabilities(seat->wlr_seat, caps);
+
+	// Hide cursor if seat doesn't have pointer capability
+	if ((caps & WL_SEAT_CAPABILITY_POINTER) == 0) {
+		wlr_cursor_set_image(seat->cursor->cursor, NULL, 0, 0, 0, 0, 0, 0);
```

Comment:
This should be `cursor_set_image(seat->cursor, NULL, NULL)`

Reply:
Changing it to that causes a segfault

Reply:
Can you change it to accept a NULL `image`?

Reply:
I had a go.

Logic seems a bit messed up anyway. I wanted to add a check to only re-show the cursor if it isn't already visible, but it doesn't work.